code-challenges.test.js

```diff
@@ -14,46 +14,272 @@
+      "Chopsticks",
+      "Mango",
+    ]);
+  });
+});
+
+// RED:
+// Good Fail: ReferenceError: capitalizeArray is not defined
 
 // b) Create the function that makes the test pass.
 
+/*
+--------- FUNCTION INFORMATION:
+Describe what the function will do: The function will accept an array, then iterate through the array to return all its' strings with the first letter capitalized.
+Function Name: capitalizeArray
+Function Parameter(s): 1, array
+--------- INPUT / OUTPUT:
+input 1: an array of strings
+output 1: an array of strings, with the first letter in each string capitalized.
+
+--------- STEPS/METHODS:
+1. Declare a function called capitalizeArray which will accept an array.
+
+2. Use .map() on the array to iterate through the elements, then use .charAt(0) and .toUpperCase() to capitalize the first letter of each string. 
+3. Use .slice() to remove the original first letter of the element that's currently being iterated, then add the end of the string to the uppercase letter.
+*/
+
+const capitalizeArray = (array) => {
+  return array.map((value) => {
+    return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
+  });
+};
+// Result: Test Suites: 1 passed, 1 total
+
+/*
+I had to do a lot of research on this one. I ended up working on the same function and forgot to comment out the failed versions, but I'll be honest here and tell you what my attempts looked like. (:
+The first time I ran it, my return looked more like this:
+// return array.map(value => value.charAt(0).toUpperCase)
+That returned ONLY the first letter of the array capitalized, which is when I realized I needed to use .slice(). So, I tried chaining .slice to the return, like this:
+// return array.map(value => value.charAt(0).toUpperCase.slice(1))
+But that returned an error specifying that it wasn't a function, so I had to do a little more research to figure out my syntax. I already had the idea to remove the first letter from the array and then put it back together somehow, but after doing research on this problem, I found this solution (code is NOT mine):
+// function capitalizeWords(arr) {
+//   return arr.map(element => {
+//     return element.charAt(0).toUpperCase() + element.slice(1).toLowerCase();
+//   });
+// }
+So, after seeing this, I realized I could use the addition operator to simply add the first letter to the latter half of the word by applying .charAt(0).toUpperCase() to the first letter, then .slice(1) to remove that letter and return the rest of the array. I also chose to leave in .toLowerCase(), because I want to stick with the idea of making the function as reusable as possible. We know the strings that we were given are all lowercase, but what if we put in strings that were all uppercase? Including .toLowerCase() will ensure that no matter what string we give this function, it will ALWAYS return that string with ONLY the first letter capitalized. I included that scenario in my test, just to confirm, and the function still passed!
```

Comment:
Excellent pseudocode and I really like that you added an edge case into your test!  Thinking outside the box and how a user _might_ break something and accounting for that in your code is going to be a very valuable skill!